scarlet/psf_match.py

```diff
@@ -1,83 +1,168 @@
+    all_params = np.array(all_params)
+
+    # Use the appropriate scheme to choose the ideal target PSF
+    # based on the best-fit parameters for each PSF
+    if extract_values is None:
+        params = []
+        if func == moffat:
+            params.append(np.mean(all_params[:,2])) # amplitude
+            params.append(np.min(all_params[:,3])*0.6) # alpha
+            params.append(np.max(all_params[:,4])*1.4) # beta
+        elif func == gaussian:
+            params.append(np.mean(all_params[:,2])) # amplitude
+            params.append(np.min(all_params[:,3])*0.6) # sigma
+        elif func == double_gaussian:
+            params.append(np.mean(all_params[:,2])) # amplitude 1
+            params.append(np.min(all_params[:,3])*0.6) # sigma 1
+            params.append(np.mean(all_params[:,4])) # amplitude 2
+            params.append(np.min(all_params[:,5])*0.6) # sigma 2
+    else:
+        params = extract_values(all_params)
+    target_psf = func(coords, y0, x0, *params)
+
+    # normalize the target PSF
+    target_psf = target_psf/np.sum(target_psf)
+    return target_psf
+
+def build_diff_kernels(psfs, target_psf, max_iter=100, e_rel=1e-3, constraints=None, cutoffs=None,
+                       l0_thresh=1e-4):
+    """Build the difference kernel to match a list of `psfs` to a `target_psf`
+
+    This convenience function runs the `Blend` class on a collection of
+    `PSFDiffKernel` objects as sources to match `psfs` to `target_psf`.
+
+    Parameters
+    ----------
+    psfs: array
+        Array of 2D psf images to match to the `target_psf`.
+    target_psf: array
+        Target PSF to use for de-convolving `psfs`.
+    max_iter: int
+        Maximum number of iterations used to create the difference kernels
+    e_rel: float
+        Relative error to use when matching the PSFs
+    constraints: `Constraint` or `ConstraintList`
+        Constraints used to match the PSFs.
+        If `constraints` is `None` then `SimpleConstraint` and `L0Constraint`
+        are used.
+    `cutoffs`: list of floats
+        Minimum non-zero value of the difference kernel for each PSF.
+        If `cutoffs` is `None`, then each PSF has no minimum value set,
+        which is the recommended value.
+
+    Returns
+    -------
+    diff_kernels: array
+        Array of 2D difference kernels for each PSF in `psfs`
+    `psf_blend`: `Blend`
+        `Blend` object that contains the result of matching
+        `psfs` to `target_psf`, where `diff_kernels` is an array
+        of the `Source.morph` for each source in `psf_blend`.
+    """
+    config = Config(refine_skip=100, source_sizes=np.array([np.max(psfs.shape[1:])]))
+    center = np.array([psfs[0].shape[0] // 2, psfs[0].shape[1] //2], dtype=psfs.dtype)
+    if constraints is None:
+        constraints = sc.SimpleConstraint() & sc.L0Constraint(l0_thresh)
+    if cutoffs is None:
+        cutoffs = [0]*len(psfs)
+    target_psfs = np.array([target_psf]*len(psfs))
```

Comment:
That is not necessary because `Gamma` in the source will expand this if it's one PSF for all bands.

Reply:
That was true, but the new Gamma has no notion about the number of bands other than from the length of `psfs`. However, I did add logic in `Source` to use a single PSF in all bands, so I removed this line.